### PR TITLE
fix(client): pass thru init error code on binary start

### LINKED ISSUE #10229: `PrismaClientInitializationError` is missing expected error code
### Bug description

When an error is caused during initialization for a PostgreSQL database, the exception `PrismaClientInitializationError` does not contain the expected error message:

```js
const { PrismaClient } = require('@prisma/client')

const prisma = new PrismaClient()

async function main() {

  try {
    const data = await prisma.$queryRaw`SELECT 1`;
  } catch (error) {
    // log error
    console.log(error)
    console.log('keys: ', Object.keys(error));
    console.log('error.errorCode: ', error.errorCode);
    console.error(JSON.stringify(error, null, 2));
  }

}

main()
  .catch((e) => {
    console.error(e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })
```

with a connection string that does not work (e.g. `postgresql://foo:bar@example.org:123/baz`) returns:

```
C:\Users\Jan\Documents\throwaway\10141>node script.js
PrismaClientInitializationError: 
Invalid `prisma.queryRaw()` invocation:


  Can't reach database server at `example.org`:`123`

Please make sure your database server is running at `example.org`:`123`.
    at cb (C:\Users\Jan\Documents\throwaway\10141\node_modules\@prisma\client\runtime\index.js:38508:17)
    at async main (C:\Users\Jan\Documents\throwaway\10141\script.js:8:18) {
  clientVersion: '3.4.2',
  errorCode: undefined
}
keys:  [ 'clientVersion', 'errorCode' ]
error.errorCode:  undefined
{
  "clientVersion": "3.4.2"
}
(node:12220) UnhandledPromiseRejectionWarning: Error: Can't reach database server at `example.org`:`123`

Please make sure your database server is running at `example.org`:`123`.
    at C:\Users\Jan\Documents\throwaway\10141\node_modules\@prisma\client\runtime\index.js:36260:21
(Use `node --trace-warnings ...` to show where the warning was created)
(node:12220) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:12220) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

(Same problem for `prisma.user.findMany({})` so this is not related to raw queries)

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: v14.17.5

### Prisma Version

3.4.2

---

Originally noticed via https://github.com/prisma/prisma/issues/10141

### PR DESCRIPTION
**Pull request by danstarns**
Passes the init error code thru when using binary engine.

Closes: https://github.com/prisma/prisma/issues/10229

### CODE REVIEW

File: packages/client/tests/functional/init-error/tests.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  () => {
+    // an example of how to query with the preloaded client
+    test('should assert that the error has the listed properties on the type', async () => {
+      expect.assertions(5)
+
+      try {
+        await prisma.$connect()
+      } catch (error) {
+        const e = error as PrismaClientInitializationError
+        expect(e.constructor.name).toEqual('PrismaClientInitializationError')
```
Comment (danstarns) [inline packages/client/tests/functional/init-error/tests.ts:18]: If i use expect().instanceOf() I get a very strange error: 

```
expect(received).toBeInstanceOf(expected)

    Expected constructor: PrismaClientInitializationError
    Received constructor: PrismaClientInitializationError
```

This is the next best thing!

Comment (millsp) [reply packages/client/tests/functional/init-error/tests.ts:18]: It's become they don't come from the same module, the generated client for the tests is bundled.

File: packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ import { dropTestSuiteDatabase, setupTestSuiteDbURI } from './setupTestSuiteEnv'
 
 export type TestSuiteMeta = ReturnType<typeof getTestSuiteMeta>
 
+export type MatrixOptions = {
+  skipDBSetup?: boolean
```
Comment (millsp) [inline packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts:8]: Maybe we can merge the two into a single `skipDb`? If you had it setup I think it's likely that you'll want teardown. wdyt?

Comment (danstarns) [reply packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts]: Sure, I have merged them!

**Review by millsp — APPROVED** [review]

**Pull request merged**